src/remote_tables/pushdown_visitor.rs

```diff
@@ -0,0 +1,204 @@
+}
+
+pub struct PostgresFilterPushdown {}
+pub struct SQLiteFilterPushdown {}
+pub struct MySQLFilterPushdown {}
+
+impl FilterPushdownVisitor for PostgresFilterPushdown {}
+
+impl FilterPushdownVisitor for SQLiteFilterPushdown {
+    fn op_to_sql(&self, op: Operator) -> Option<String> {
+        match op {
+            Operator::RegexMatch
+            | Operator::RegexIMatch
+            | Operator::RegexNotMatch
+            | Operator::RegexNotIMatch
+            | Operator::BitwiseXor => None,
+            _ => Some(op.to_string()),
+        }
+    }
+}
+
+impl FilterPushdownVisitor for MySQLFilterPushdown {
+    fn op_to_sql(&self, op: Operator) -> Option<String> {
+        match op {
+            // TODO: see if there's a way to convert the non-case sensitive match
+            Operator::RegexIMatch | Operator::RegexNotIMatch => None,
+            Operator::RegexMatch => Some("RLIKE".to_string()),
+            Operator::RegexNotMatch => Some("NOT RLIKE".to_string()),
+            Operator::BitwiseXor => Some("^".to_string()),
+            _ => Some(op.to_string()),
+        }
+    }
+}
+
+pub trait FilterPushdownVisitor {
+    fn scalar_value_to_sql(&self, value: &ScalarValue) -> Option<String> {
+        match value {
+            ScalarValue::Utf8(Some(val)) => Some(format!("'{}'", val)),
+            _ => Some(format!("{}", value)),
+        }
+    }
+
+    fn op_to_sql(&self, op: Operator) -> Option<String> {
+        Some(op.to_string())
+    }
+}
+
+impl<T: FilterPushdownVisitor> ExpressionVisitor for FilterPushdown<T> {
+    fn pre_visit(mut self, expr: &Expr) -> Result<Recursion<Self>> {
+        match expr {
+            Expr::Column(_) | Expr::Literal(_) => {}
+            Expr::BinaryExpr(BinaryExpr { op, .. }) => {
+                // Check if operator pushdown supported; left and right expressions will be checked
+                // through further recursion.
+                if self.source.op_to_sql(*op).is_none() {
+                    return Ok(Recursion::Stop(self));
+                }
+            }
+            _ => {
+                // Expression is not supported, no need to visit any remaining nodes
+                self.pushdown_supported = false;
+                return Ok(Recursion::Stop(self));
+            }
+        };
+        Ok(Recursion::Continue(self))
+    }
+
+    fn post_visit(mut self, expr: &Expr) -> Result<Self> {
+        match expr {
+            Expr::Column(col) => self.sql_exprs.push(col.name.clone()),
+            Expr::Literal(val) => {
+                let sql_val = self.source.scalar_value_to_sql(val).ok_or_else(|| {
+                    DataFusionError::Execution(format!(
+                        "Couldn't convert ScalarValue {:?} to a compatible one for the remote system",
+                        val,
+                    ))
+                })?;
+                self.sql_exprs.push(sql_val)
+            }
+            Expr::BinaryExpr(be @ BinaryExpr { .. }) => {
```

Comment:
This is out of scope for this PR, but I think eventually we'll want to support function pushdown. Something like `date_trunc` / `date_part` is very useful in reporting queries (e.g. I can get data for a certain month) but won't be pushed down right now.

Reply:
Yup, there's a bunch of other expressions that are not ship-able atm, such as `IsNull`, `InList`, `Exists` and other `Expr` enum variants. The plan is to iteratively add support for those eligible in subsequent PRs.